### PR TITLE
docs: Adding canonical links

### PR DESCRIPTION
**Pull request by thomasgilmore95**
Adding canonical links to blog posts. This pull request solves issue #568

### CODE REVIEW

File: _posts/2021-10-22-eslint-v8.1.0-released.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ tags:
   - release
   - minor
 author: mdjermanovic
+original:
```
Comment (nzakas) [inline _posts/2021-10-22-eslint-v8.1.0-released.md:8]: We would not include this section on all blog posts, only the ones that were re-posted from elsewhere.

**Comment by thomasgilmore95** [comment]
I'm not sure how to fix the commit message error. I've changed the PR Title and changed the commit message.

File: _posts/2021-10-22-eslint-v8.1.0-released.md (hunk 1 above)
Comment (thomasgilmore95) [reply _posts/2021-10-22-eslint-v8.1.0-released.md]: I'll remove this section from the blog post

**Comment by aladdin-add** [comment]
@thomasgilmore95 we have started to use the conventional commits. https://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes

**Comment by aladdin-add** [comment]
no worries, I've fixed the title.

**Comment by thomasgilmore95** [comment]
@aladdin-add thank you

**Review by nzakas — COMMENTED** [review]
This should work. Can you add the blog post I left the gist for so we can see it in action via the deploy preview.

**Comment by thomasgilmore95** [comment]
I added back to blog posts to see it in deploy preview.

**Comment by nzakas** [comment]
Looks like it's not working:
https://deploy-preview-885--eslint.netlify.app/blog/2018/02/the-inception-of-eslint

The canonical URL isn't the original URL.

**Comment by thomasgilmore95** [comment]
I tested the canonical url and it comes back `href(unknown)`. I made a commit so you could see it as well.

**Comment by thomasgilmore95** [comment]
Is there another way to access the page's front matter variables.

**Comment by nzakas** [comment]
I don’t know Eleventy well enough to answer that. I’d suggest checking the docs: https://www.11ty.dev/docs/data-frontmatter/

**Comment by thomasgilmore95** [comment]
Thank you that link was helpful. The canonical link is being show know if original is present.

File: _includes/head.liquid [hunk 2]
```diff
@@ -26,7 +26,11 @@
     <title>{{ site.title }}</title>
   {% endif %}
   <link rel="preconnect" href="https://www.google-analytics.com">
-  <link rel="canonical" href="{{ page.url | replace: '.html', '' }}">
+  {% if original %} 
+    <link rel="canonical" href="{{ original.url }}">
```
Comment (nzakas) [inline _includes/head.liquid:30]: 👌

**Review by nzakas — COMMENTED** [review]
Nice work! A couple of finishing touches:

1. When there is an `original`, can you add a paragraph at the top of the post that says "This post originally appeared on {{original.site.name}} on {{original.date}}." where the site name is linked to original.site.url and the date is linked to original.url
2. Can up update the date in the blog post filename to be 2021-11-09? We want it to appear at the top of the most recent posts, not when it originally was posted three years ago.

Thanks!

**Comment by thomasgilmore95** [comment]
Please let me know if you want me to make any changes.

File: _layouts/post.liquid [hunk 3]
```diff
@@ -28,6 +28,11 @@
               <em> - </em>
             {% endif %}<em>{{ page.date | dateFormat: "d LLLL yyyy" }}</em>
             <br />
+            {% if original %}
+            <br />
+              This post originally appeared on {{ original.site.name }} on {{ original.date | dateFormat: "d LLLL yyyy" }}.
```
Comment (nzakas) [inline _layouts/post.liquid:33]:
```suggestion
              This post originally appeared on <a href=“{{original.site.url}}”>{{ original.site.name }}</a> on <a href=“{{original.url}}”>{{ original.date | dateFormat: "d LLLL yyyy" }}</a>.
```

**Review by nzakas — APPROVED** [review]

**Review by nzakas — APPROVED** [review]
LGTM. Thanks for sticking with this. We will look at merging next week.

**Comment by thomasgilmore95** [comment]
Thank you for letting me work on this.

**Review by nzakas — APPROVED** [review]

**Pull request merged**